mobilecoind/src/bin/main.rs

```diff
@@ -106,30 +110,71 @@ fn main() {
 
             let _ = std::fs::create_dir_all(mobilecoind_db);
 
-            let mobilecoind_db = Database::new(mobilecoind_db, logger.clone())
-                .expect("Could not open mobilecoind_db");
+            // This if contains a lot of duplication due to TransactionManager and Service being
```

Comment:
```suggestion
            // This contains a lot of duplication due to TransactionManager and Service being
```

Reply:
alternate suggestion: `This if-else contains`